json.md

```diff
@@ -0,0 +1,301 @@
+
+Manual JSON serialization refers to using the built-in JSON decoder in `dart:convert`. It involves passing the raw JSON string to the `JSON.decode()` method, and then looking up the values you need in the `Map<String, dynamic>` the method returns. It involves no external dependencies or particular setup process, and it is good for quick proof of concepts.
+
+Where the manual serialization does not perform well is when your project becomes bigger. Writing the serialization logic by hand can become hard to manage and error-prone. If you have a typo when accessing an unexisting JSON field, your code throws an error during runtime.
+
+If you do not have many JSON models in your project and are looking to test a concept quickly, manual serialization might be the way you want to start.
+
+### Use code generation for medium to large projects 
+
+JSON serialization with code generation means having an external library generate the serialization boilerplate for you. They involve some initial setup and running a file watcher that generates the code from your model classes. For example, [json_serializable](https://pub.dartlang.org/packages/json_serializable) and [built_value](https://pub.dartlang.org/packages/built_value) are these kind of libraries.
+
+This approach scales well for a larger project. There are no hand-written boilerplate needed, and typos when accessing JSON fields are caught at compile-time. The downside with code generation is that it involves some initial setup. Also, the generated source files may produce visual clutter in your project navigator
+
+You might want to use generated code for JSON serialization when you have a medium or a larger project.
+
+## Is there a GSON/Jackson/Moshi equivalent in Flutter?
+
+Dart has supported _tree shaking_ for quite a long time. With tree shaking, we can “shake off” unused code from our release builds. Tree shaking allows us to optimize the size of our applications significantly.
+
+While Dart users might have gotten used to [dartson](https://pub.dartlang.org/packages/dartson) to (de)serialize their JSON, it does not work with Flutter. The reason behind this is that `dartson`, like [GSON](https://github.com/google/gson) in Java, uses _runtime reflection_. In Flutter, reflection (also known as `dart:mirrors`) is disabled.
+
+Since reflection makes all code implicitly used by default, it interferes with tree shaking. The tools cannot know what parts are unused at runtime; the redundant code is impossible to strip away. App sizes cannot be optimized when using reflection.
+
+Although we cannot use dartson, there’s a library that gives us a simple to use API, without using reflection. That library is called [json_serializable](https://pub.dartlang.org/packages/json_serializable).
+
+## Serializing JSON manually using dart:convert
+
+Basic JSON serialization in Flutter is very simple. Dart comes with the `dart:convert` library built in, and using the JSON decoder API is straightforward.
```

Comment:
"comes with" and "built in" are redundant

Reply:
maybe: "Flutter has a built-in 'dart:convert' library, which includes a straightforward JSON encoder and decoder."